src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,223 @@
+   * with the very first timestamp. If not, return a tracker with end time.
+   *
+   * @param startTimeTrackingMilliseconds - the start time for tracking, epoch milliseconds UTC
+   * @param includeTagGroupA include tag group A
+   * @param includeTagGroupB include tag group B
+   * @param includeTagGroupC include tag group C
+   * @param includeTagGroupD include tag group D
+   * @param stringHistorical export string historical logs if true
+   * @return {@link RapidCatchUpTrack} object
+   * @throws Exception for {@link DataPoint} getTimeStampAsDate() errors
+   * @throws IOException for errors closing exporter
+   * @since 1.14.2
+   */
+  public static RapidCatchUpTrack rapidCatchUpRequest(
+      long startTimeTrackingMilliseconds,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD,
+      boolean stringHistorical)
+      throws Exception, IOException {
+
+    // the end time is the start + catch up duration
+    long endTimeTrackingMilliseconds =
+        catchUpRequestDurationMilliseconds + startTimeTrackingMilliseconds;
+
+    final String ebdStartTime =
+        HistoricalDataQueueManager.convertToEBDTimeFormat(startTimeTrackingMilliseconds);
+    final String ebdEndTime =
+        HistoricalDataQueueManager.convertToEBDTimeFormat(endTimeTrackingMilliseconds);
+
+    final String ebdRequest =
+        HistoricalDataManager.prepareHistoricalFifoReadEBDString(
+            ebdStartTime,
+            ebdEndTime,
+            includeTagGroupA,
+            includeTagGroupB,
+            includeTagGroupC,
+            includeTagGroupD,
+            stringHistorical);
+
+    RapidCatchUpTrack histTracker;
+    try {
+      final Exporter exporter = HistoricalDataManager.executeEbdCall(ebdRequest);
+      histTracker =
+          parseEBDHistoricalLogExportResponseRapidCatchUp(exporter, endTimeTrackingMilliseconds);
+      if (histTracker.isHistoricalTrackingCaughtUp()) {
+        // if caught up, reset the period
+        resetRapidCatchUpPeriod();
+      } else {
+        // if not, change period
+        adjustRapidCatchUpPeriod();
+      }
+    } catch (EbdTimeoutException e) {
+      System.out.println(
+          "EBD timeout Exception during rapid catch up, will reset catch up period.");
+      // If the base period is causing timeout, drop all the way down to 1 minute
+      if (catchUpRequestDurationMilliseconds == RAPID_CATCH_UP_TIME_BASE) {
+        catchUpRequestDurationMilliseconds = 1000 * 60;
+      } else {
+        resetRapidCatchUpPeriod();
+      }
+      histTracker = new RapidCatchUpTrack(false, startTimeTrackingMilliseconds);
+    }
+
+    return histTracker;
+  }
+
+  /**
+   * Parses Export Block Descriptor Historical Log response into {@link DataPoint}, expects response
+   * to be empty. This happens when the Flexy has been powered off. Note: this function only handles
```

Comment:
the "expects response to be empty" bit is a little confusing. I think i see the intention, that you would expect for most calls the EBD historical log call to be empty as you are in the process of parsing through time that the Flexy was offline for. but using the work expects makes it sound like the case where you do find a datapoint would be "unexpected" when in reality you have just reached the end of that empty time block.

Reply:
I will change up the wording. Perhaps this is an indication that the function tries to do too much.